users/serializers.py

```diff
@@ -1,72 +1,148 @@
+                raise serializers.ValidationError('Email already exists!')
+            return email
+        raise serializers.ValidationError('Invalid email format!')
+    
+    
     def create(self, validated_data):
-        validated_data["is_active"] = True
-        user = UserAccount.objects.create_user(**validated_data)
+        password = self.validated_data.get('password')
+        # pattern = r'!@#$%&\^*(),.?":{}|<>'
+        
+        error_messages = []
+        
+        # if not re.search(pattern, password):
+        #         error_messages.append(f'Passwords must contain at least one SPECIAL characters {pattern}')
+            
+        try: 
+            validate_password(password)   
+        except serializers.ValidationError as e:
+            error_messages.append(str(e))
+        
+        if error_messages:
+            raise serializers.ValidationError(error_messages)
+        
+        user = Account.objects.create_user(**validated_data)
         return user
 
-    # def update(self, instance, validated_data):
-    #     instance.email = validated_data.get('new_email', instance.email)
-    #     instance.username = validated_data.get('new_username', instance.username)
 
-    #     instance.save()
-    #     return instance 
+class ListAccountsSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Account
+        fields = '__all__'
+
 
 
 class LoginSerializer(serializers.Serializer):
     username_or_email = serializers.CharField(required=True)
     password = serializers.CharField(required=True)
 
-    class Meta:
-        fields = ["username_or_email", "password"]
+    def validate(self, data):
+        pattern = '[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+'
+        username_or_email = data.get('username_or_email')
+        password = data.get('password')
+        
+        if re.match(pattern, username_or_email):
+            try:
+                user = Account.objects.get(email=username_or_email)
+                username_or_email_field = 'email'
+            except Account.DoesNotExist:
+                raise serializers.ValidationError('Account with Email does not exist!')
+        else:
+            try:
+                user = Account.objects.get(username=username_or_email)
+                username_or_email_field = 'username'
+            except Account.DoesNotExist:
+                raise serializers.ValidationError('Account with Username does not exist!')
+        
+        if not user.check_password(password):
+            raise serializers.ValidationError('Invalid password!')
+        
+        data['user'] = user
+        data['username_or_name'] = username_or_email_field
+        return data
+        
+
+
 
 
 class ResetPasswordSerializer(serializers.Serializer):
     old_password = serializers.CharField(required=True)
     new_password = serializers.CharField(required=True)
     confirm_password = serializers.CharField(required=True)
```

Comment:
Let the frontend handle confirmation of password. You only have to handle old and new passwords